pkg/etcdutil/etcdutil.go

```diff
@@ -248,25 +248,31 @@ func GetEtcdEndPointsSorted(ctx context.Context, clientMaintenance client.Mainte
 
 // TakeAndSaveFullSnapshot takes full snapshot and save it to store
 func TakeAndSaveFullSnapshot(ctx context.Context, client client.MaintenanceCloser, store brtypes.SnapStore, lastRevision int64, cc *compressor.CompressionConfig, suffix string, isFinal bool, logger *logrus.Entry) (*brtypes.Snapshot, error) {
+	startTime := time.Now()
```

Comment:
In general this is fine. If you wish to improve the readability of the code then you can do the following (I am assuming that you will need to measure time in a LOT of places):

- Break down this function into 3 parts which can then be independently tested (unit/live)
- Introduce new utility methods:
```go
func measureRunTime(name string) func() {
     start := time.Now()
     return dumpRunTime(name, start)
}

func dumpRunTime(name string, start time.Time) func() {
       return func() {
             logger.Infof("Total run time for %s: is %f seconds", time.Since(start).Seconds())
       }
}
```
You can improve it by passing in the `sink` for the logs to the `measureRunTime` but right not you are only printing the logs to stdout so not an issue.

To use the utility function all you have to do is in the beginning of your function have:
```go
defer measureRunTime(<name>)()
```

This is not a mandatory thing to do and therefore even without this change you can merge this PR.

Reply:
hmm ... make sense to me.
We can take this as an improvement later as we need to update the time with in lot of places.

Reply:
right now, I'm merging the PR without this change as we need to release thee backup-restore today hope thats fine.

Reply:
Yes as i said this is not a mandatory change (only good to have). Please go ahead with the merge.